File: java/FilmLayarLebar.java
// Sub-child Class 
public class FilmLayarLebar extends Film {
    // Atribut khusus bioskop
    private double hargaTiket;
    private String jaringanBioskop;
    private String formatProyeksi;

    // Constructor penuh memanggil constructor Film
    public FilmLayarLebar(String idMedia, String judul, int tahunRilis, String genre, String sutradara, 
                          String ratingUsia, double hargaTiket, String jaringanBioskop, String formatProyeksi) {
        super(idMedia, judul, tahunRilis, genre, sutradara, ratingUsia); // Memanggil constructor Film
        this.hargaTiket = hargaTiket;
        this.jaringanBioskop = jaringanBioskop;
        this.formatProyeksi = formatProyeksi;
    }

    // Setter
    public void setHargaTiket(double hargaTiket) { this.hargaTiket = hargaTiket; }
    public void setJaringanBioskop(String jaringanBioskop) { this.jaringanBioskop = jaringanBioskop; }
    public void setFormatProyeksi(String formatProyeksi) { this.formatProyeksi = formatProyeksi; }

    // Getter
    public double getHargaTiket() { return hargaTiket; }
    public String getJaringanBioskop() { return jaringanBioskop; }
    public String getFormatProyeksi() { return formatProyeksi; }
}

File: cpp/FilmLayarLebar.cpp
#include "Film.cpp"

// Sub-child Class 
class FilmLayarLebar : public Film
{
private:
    // Atribut khusus komersil bioskop
    double hargaTiket;
    string jaringanBioskop;
    string formatProyeksi;

public:
    // Constructor penuh memanggil constructor Film
    FilmLayarLebar(string idMedia, string judul, int tahunRilis, string genre, string sutradara, string ratingUsia, double hargaTiket, string jaringanBioskop, string formatProyeksi)
        : Film(idMedia, judul, tahunRilis, genre, sutradara, ratingUsia) // Inisialisasi atribut Film & KaryaMedia
    {
        this->hargaTiket = hargaTiket;
        this->jaringanBioskop = jaringanBioskop;
        this->formatProyeksi = formatProyeksi;
    }

    // Setter atribut FilmLayarLebar
    void setHargaTiket(double hargaTiket) { this->hargaTiket = hargaTiket; }
    void setJaringanBioskop(string jaringanBioskop) { this->jaringanBioskop = jaringanBioskop; }
    void setFormatProyeksi(string formatProyeksi) { this->formatProyeksi = formatProyeksi; }

    // Getter atribut FilmLayarLebar
    double getHargaTiket() const { return hargaTiket; }
    string getJaringanBioskop() const { return jaringanBioskop; }
    string getFormatProyeksi() const { return formatProyeksi; }

    // Destructor
    ~FilmLayarLebar() {}
};